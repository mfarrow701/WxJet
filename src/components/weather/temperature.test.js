import React from 'react';
import {shallow} from 'enzyme';
import Temperature from './temperature';

describe('Temperature component', () => {
    let component;

    const renderWrapper = () => {
        component = shallow(<Temperature/>);
    };

    beforeEach(() => {
        renderWrapper();
    });

    it('should render the temperature component after load', () => {
        expect(component).toMatchSnapshot();
        expect(component.find('.Temperature').text()).toEqual('--째C');
    });

    it('should render the temperature component with a custom value', () => {
        component.setProps({value: 7});
        expect(component.find('.Temperature').text()).toEqual('7째C');
    });

    it('should toggle between celsius and fahrenheit on click', () => {
        component.setProps({value: 7});
        component.find('.Temperature').simulate('click');
        expect(component.find('.Temperature').text()).toEqual('44.6째F');
        component.find('.Temperature').simulate('click');
        expect(component.find('.Temperature').text()).toEqual('7째C');
    });
});
