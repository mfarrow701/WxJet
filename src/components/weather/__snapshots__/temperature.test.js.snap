// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temperature component should render the temperature component after load 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Temperature />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "--째C",
      "className": "Temperature",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "--째C",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "--째C",
        "className": "Temperature",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "--째C",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
